BUILD.md

```diff
@@ -0,0 +1,69 @@
+# Building DLAF
+
+## Dependencies
+
+- MPI
+- [pika](https://github.com/pika-org/pika)
+- [umpire](https://github.com/LLNL/Umpire)
+- [blaspp](https://bitbucket.org/icl/blaspp/src/default/)
+- [lapackpp](https://bitbucket.org/icl/lapackpp/src/default/)
+- Intel MKL or other LAPACK implementation
+- A ScaLAPACK implementation (optional, ScaLAPACK-like C API only)
+- [whip](https://github.com/eth-cscs/whip) (optional, GPU only)
+- [cuda](https://developer.nvidia.com/cuda) (optional, NVidia GPUs only)
+- [HIP/ROCm](https://github.com/RadeonOpenCompute/ROCm) (optional, AMD GPUs only)
+- [GoogleTest](https://github.com/google/googletest) (optional; bundled) - unit testing
+- Doxygen (optional) - documentation
+
+## Build using Spack
+
+We provide a spack package `dla-future` that can be easily added to your own spack as follows:
+
+`spack repo add $DLAF_ROOT/spack`
+
+This will add a new repository with namespace `dlaf`.
+
+Example installation:
+
+`spack install dla-future ^[virtuals=blas,lapack] intel-oneapi-mkl`
+
+or you can also build C API compatible with ScaLAPACK with:
+
+`spack install dla-future +scalapack ^[virtuals=blas,lapack,scalapack] intel-oneapi-mkl`
+
+You can go even further with a more detailed spec like this one, which builds dla-future in debug mode, using the clang compiler, specifying that the pika on which it depends has to be built
+in debug mode too, and that we want to use MPICH as MPI implementation, without fortran support (because clang does not support it).
+
+`spack install dla-future %clang build_type=Debug ^pika build_type=Debug ^mpich ~fortran`
+
+## Build the old good way
+
+You can build all the dependencies by yourself, but you have to ensure that:
+- BLAS/LAPACK implementation is not multithreaded
```

Comment:
This isn't true anymore, right? Would it make sense to separate the README changes to a new PR?

Reply:
#1138